Services/communication_service.c

```diff
@@ -19,13 +19,33 @@
 
 #include "communication_service.h"
 
+#include <FreeRTOS.h>
+#include <csp/csp.h>
+#include <csp/csp_endian.h>
+
+#include "comms_hal.h"
+#include "service_response.h"
+#include "service_utilities.h"
+#include "services.h"
+#include "system.h"
+#include "uhf_hal.h"
+
+#define CHAR_LEN 4  // Numpy unicode string character length
+#define CALLSIGN_LEN 6
+#define BEACON_MSG_LEN_MAX 36
+#define FRAM_SIZE 16
+#define SID_byte 1
+
+static Sband_config S_config;  // In two different situations, globalization has
```

Comment:
Any chance this could be not global too?

Reply:
No, the cnv function for frequency still crashes, and I'm out of ideas. For now, I'd say let's keep it global until we figure it out. Btw I noticed that all the structs in HALs are global. Is that ok?

Reply:
I could percieve a need to have them be persistent - perhaps if we want to return the most recent value of something if you couldn't retreive the value from hardware or something. But we will have to go through in improve the HAL to make it thread-safe. We can leave it for now as a first sweep.